src/joystick/linux/SDL_sysjoystick.c

```diff
@@ -708,7 +708,7 @@ static void LINUX_FallbackJoystickDetect(void)
 
             count = scandir("/dev/input", &entries, filter_entries, NULL);
             if (count > 1) {
-                qsort(entries, count, sizeof(*entries), sort_entries);
+                SDL_qsort(entries, count, sizeof(*entries), sort_entries);
```

Comment:
For correctness, you should mark `sort_entries` as `SDLCALL` after this.

Besides: What is actually the point here where you are actually calling `scandir` from libc?

Reply:
ok. I ll fix that later.

> Besides: What is actually the point here where you are actually calling scandir from libc?

I didn't write this. but try ... https://github.com/libsdl-org/SDL/issues/7825 
good point to see the "scandir" .. no sdl equivalent :/

Reply:
ok done https://github.com/libsdl-org/SDL/commit/6f344f29b84685ce4704809d733dcb52186e600d